src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2365,36 +2365,22 @@ private void LexSyntaxTrivia(bool afterFirstToken, bool isTrailing, ref SyntaxLi
 
         private bool IsConflictMarkerTrivia()
         {
-            var position = TextWindow.Position;
-            var text = TextWindow.Text;
-            if (position == 0 || SyntaxFacts.IsNewLine(text[position - 1]))
+            // Is directly after a newline?
+            if (!IsAtNewLine())
             {
-                var firstCh = text[position];
-                Debug.Assert(firstCh == '<' || firstCh == '=' || firstCh == '>');
-
-                if ((position + s_conflictMarkerLength) <= text.Length)
-                {
-                    for (int i = 0, n = s_conflictMarkerLength; i < n; i++)
-                    {
-                        if (text[position + i] != firstCh)
-                        {
-                            return false;
-                        }
-                    }
-
-                    if (firstCh == '=')
-                    {
-                        return true;
-                    }
-
-                    return (position + s_conflictMarkerLength) < text.Length &&
-                        text[position + s_conflictMarkerLength] == ' ';
-                }
+                return false;
             }
-
-            return false;
+            // Is the branch seperator marker?
+            if (TextWindow.NextAre("======="))
+            {
+                return true;
+            }
+            // Is branch conflict marker, and followed by a space.
+            return TextWindow.NextIs(' ', s_conflictMarkerLength) && (TextWindow.NextAre("<<<<<<<") || TextWindow.NextAre(">>>>>>>"));
```

Comment:
this seems really counterintuitive to how i woudl expect to check.  Why not just check for `"<<<<<<< "` or `">>>>>>> "`?

Reply:
this feels like the entire method should just be:  `IsAfterNewLine() && (TextWindow.NextAre("=======") || TextWindow.NextAre("<<<<<<< ") || TextWindow.NextAre(">>>>>>> ")`

Reply:
Updated to use simpler implementation.